[user] Add getSites functions in User class

CamilleBeau opened the pull request:
## Brief summary of changes
This PR adds 2 functions to the user class. One returns an array of Site objects that the user has access to. The other returns an associative array with the format CenterID => SiteName for all of the sites that the user has access to.

- [x] Have you updated related documentation?

#### Testing instructions (if applicable)

#### Link(s) to related issue(s)

jeffersoncasimir commented on modules/user_accounts/php/user_accounts.class.inc:
```diff
@@ -77,9 +77,24 @@ class User_Accounts extends \DataFrameworkMenu
+        if ($user->hasPermission('user_accounts_multisite')) {
+            $sites = \Utility::getSiteList(false);
+        } else {
+            $sites = [];
```
I believe you should be using `$user->getStudySites()` instead of reimplementing the logic

CamilleBeau replied:
@jeffersoncasimir getStudySites only includes study sites, but I believe in this case we want _all_ sites that the user has access to

CamilleBeau commented:
@jeffersoncasimir @driusan (and @ridz1208 because you seem interested in this development) I added 2 functions to the User class: 
1. getSites() that returns an array of Site objects for all sites that the user has
2. getSitesNamesList() that returns an associative array of centerID => centerName

I'm still trying to set up Docker on my VM to be able to debug the test suites (it was failing yesterday), but I'm trying something out in my last commit.

jeffersoncasimir reviewed (approved):
This has been tested and works as expected. Great work!

driusan commented on modules/user_accounts/php/useraccountrowprovisioner.class.inc:
```diff
@@ -17,6 +17,18 @@ class UserAccountRowProvisioner extends \LORIS\Data\Provisioners\DBRowProvisione
+        } else {
+            $siteLimitQuery = '';
+            $params         = [];
+        }
```
I don't understand why this is necessary.

user_accounts.class.inc already has:
```
    /**
     * Tells the base class that this page's provisioner can support the
     * HasAnyPermissionOrUserSiteMatch filter.
     *
     * @return ?array of permissions or null
     */
    public function allSitePermissionNames() : ?array
    {
        return ['user_accounts_multisite'];
    }
```

so shouldn't the HasAnyPermissionOrUserSiteMatch filter take care of it?

CamilleBeau commented:
@jeffersoncasimir @driusan I removed all of the user_accounts code, changed the name, and the description. The PR now only includes the 2 functions that can be used elsewhere in the code.
